server/src/main/java/org/spine3/server/entity/FieldMasks.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2016, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.entity;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.FieldMask;
+import com.google.protobuf.Message;
+import com.google.protobuf.ProtocolStringList;
+import org.spine3.protobuf.KnownTypes;
+import org.spine3.protobuf.TypeUrl;
+
+import javax.annotation.Nullable;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A utility class for {@code FieldMask} processing against instances of {@link Message}.
+ *
+ * @author Dmytro Dashenkov
+ */
+@SuppressWarnings("UtilityClass")
+public class FieldMasks {
+
+    private FieldMasks() {
+    }
+
+    /**
+     * <p>Applies given {@code FieldMask} to given collection of {@link Message}s.
+     * Does not change the {@link Collection} itself.
+     *
+     * <p>The {@code FieldMask} must be valid for this operation.
+     *
+     * <p>In case of failure (e.g. wrong {@code TypeUrl} in the field mask) this method silently handles it
+     * without any exception or error thrown.
+     *
+     * @param mask     {@code FieldMask} to apply to each item of the input {@link Collection}.
+     * @param entities {@link Message}s to filter.
+     * @param typeUrl  Type of the {@link Message}s.
+     * @return Non-null unmodifiable {@link Collection} of {@link Message}s of the same type that the input had.
+     * @see #isValid(FieldMask)
+     */
+    @SuppressWarnings({"MethodWithMultipleLoops", "unchecked"})
+    public static <M extends  Message, B extends Message.Builder> Collection<M> applyMask(@SuppressWarnings("TypeMayBeWeakened") FieldMask mask, Collection<M> entities, TypeUrl typeUrl) {
+        final List<M> filtered = new ArrayList<>();
```

Comment:
Let's use `LinkedList` here as well.

You only use `ArrayList` in case 

-  you know the size exactly
or
- you expect heavy `get(index)` usage.